infra/growth/src/util/sdnMatcher.js

```diff
@@ -0,0 +1,99 @@
+const fs = require('fs')
+const logger = require('../logger')
+
+const sdnCsvFilename = `${__dirname}/../../data/sdn.csv`
+
+class CsvLineParser {
+  constructor() {
+    this.reValid = /^\s*(?:'[^'\\]*(?:\\[\S\s][^'\\]*)*'|"[^"\\]*(?:\\[\S\s][^"\\]*)*"|[^,'"\s\\]*(?:\s+[^,'"\s\\]+)*)\s*(?:,\s*(?:'[^'\\]*(?:\\[\S\s][^'\\]*)*'|"[^"\\]*(?:\\[\S\s][^"\\]*)*"|[^,'"\s\\]*(?:\s+[^,'"\s\\]+)*)\s*)*$/
+    this.reValue = /(?!\s*$)\s*(?:'([^'\\]*(?:\\[\S\s][^'\\]*)*)'|"([^"\\]*(?:\\[\S\s][^"\\]*)*)"|([^,'"\s\\]*(?:\s+[^,'"\s\\]+)*))\s*(?:,|$)/g
+  }
+
+  /**
+   * Parses a csv line and returns an array.
+   *
+   * @param line
+   * @returns {Array<string>}
+   */
+  parse(line) {
+    // Return NULL if input string is not well formed CSV string.
+    if (!this.reValid.test(line)) return null
+    // Initialize array to receive values.
+    const a = []
+    // Walk the string using replace with callback.
+    line.replace(this.reValue, function(m0, m1, m2, m3) {
+      // Remove backslash from \' in single quoted values.
+      if (m1 !== undefined) a.push(m1.replace(/\\'/g, "'"))
+      // Remove backslash from \" in double quoted values.
+      else if (m2 !== undefined) a.push(m2.replace(/\\"/g, '"'))
+      else if (m3 !== undefined) a.push(m3)
+      return ''
+    })
+    // Handle special case of empty last value.
+    if (/,\s*$/.test(line)) a.push('')
+    return a
+  }
+}
+
+class SdnMatcher {
+  constructor() {
+    this.lastNames = {} // Maps lastname -> Array<firstName>
+    this._parseData(new CsvLineParser())
+  }
+
+  /**
+   * Reads the CDN csv files and populates the lastNames structure.
+   * @private
+   */
+  _parseData(lineParser) {
+    const data = fs.readFileSync(sdnCsvFilename).toString()
+    const lines = data.split('\r\n')
+    for (const line of lines) {
+      const cells = lineParser.parse(line)
+      if (!cells || cells.length < 2) {
+        continue
+      }
+      if (cells[2] !== 'individual') {
+        continue
+      }
+      const values = cells[1].split(',').map(s => s.toLowerCase().trim())
+      const firstName = values[0]
+      const lastName = values[1]
+      if (!firstName || !lastName) {
+        continue
+      }
+
+      if (this.lastNames[lastName]) {
+        this.lastNames[lastName].push(firstName)
+      } else {
+        this.lastNames[lastName] = [firstName]
+      }
+    }
+    logger.info(
+      `Read ${Object.keys(this.lastNames).length} records from SDN list.`
+    )
+  }
+
+  /**
+   * Returns true if the first+last name match an individual in the SDN list.
+   *
+   * @param matcher
+   * @param firstName
+   * @param lastName
+   * @returns {boolean}
+   */
+  match(firstName, lastName) {
+    // Normalize the input.
+    const normFirstName = firstName.trim().toLowerCase()
+    const normLastName = lastName.trim().toLowerCase()
+
+    // Check against records of lastNames/firstNames.
+    const firstNames = this.lastNames[normLastName]
```

Comment:
I was just checking some balkan names and people have local letter variants that are simplified in the SDN document. For example a serb would put his name in as `Miroslav Radić` while in the SDN document it would show up as `Miroslav Radic` -> notice the different `c` letters.

There are probably similar simplifications od spanish `ñ` and german `ä`, `ö`, `ü` and a ton of other letters. 

I think we should simplify the users input `name` + `lastName` before we do the matching.

Reply:
Yes, good points. There are lot of things we could do to improve the matching... Fuzzy matching would handle the case you mentioned. So perhaps we can focus on that first. Would you be ok if I add a TODO and land this as is ? Then we can iterate to make this better if needed...

Reply:
Yup that sounds good